Java 21 Build Fails

```
[ERROR] Errors: 
[ERROR]   StateMachineServiceTest.testChangeMachineStatesInBatch_noDocuments:65 » IllegalState java.lang.IllegalStateException: Illegal pop() with non-matching JdbcValuesSourceProcessingState
[ERROR]   StateMachineServiceTest.testChangeMachineStatesInBatch_submitting:57 » IllegalState Illegal pop() with non-matching JdbcValuesSourceProcessingState
```

--- a/.github/workflows/codeql-analysis.yml
+++ b/.github/workflows/codeql-analysis.yml
@@ -19,3 +19,4 @@ jobs:
       # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
       # Use only 'java' to analyze code written in Java, Kotlin or both
       # Use only 'javascript' to analyze code written in JavaScript, TypeScript or both
+      java_version: 21

--- a/.github/workflows/coverity-scan.yml
+++ b/.github/workflows/coverity-scan.yml
@@ -14,3 +14,4 @@ jobs:
       project-name: ${{ github.event.repository.name }}
       version: ${{ github.sha }}
       description: ${{ github.ref }}
+      java_version: 21

--- a/.github/workflows/maven-deploy.yml
+++ b/.github/workflows/maven-deploy.yml
@@ -34,6 +34,7 @@ jobs:
     with:
       environment: internal-publish
       release_type: snapshot
+      java_version: 21
     secrets:
       username: ${{ secrets.MAVEN_CENTRAL_USERNAME }}
       password: ${{ secrets.MAVEN_CENTRAL_PASSWORD }}
@@ -45,6 +46,7 @@ jobs:
     with:
       environment: ${{ inputs.environment }}
       release_type: ${{ inputs.release_type }}
+      java_version: 21
     secrets:
       username: ${{ secrets.MAVEN_CENTRAL_USERNAME }}
       password: ${{ secrets.MAVEN_CENTRAL_PASSWORD }}

--- a/.github/workflows/maven-test.yml
+++ b/.github/workflows/maven-test.yml
@@ -16,3 +16,5 @@ jobs:
   maven-tests:
     uses: wultra/wultra-infrastructure/.github/workflows/maven-test.yml@develop
     secrets: inherit
+    with:
+      java_version: 21

--- a/enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/IdentityVerificationRepository.java
+++ b/enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/IdentityVerificationRepository.java
@@ -55,28 +55,36 @@ public interface IdentityVerificationRepository extends CrudRepository<IdentityV
     /**
      * Return all identity verifications eligible for change to next state.
      *
+     * @param documentVerificationProvider Document verification production name.
      * @return identity verifications
      */
-    @Query("SELECT id FROM IdentityVerificationEntity id WHERE" +
-            " (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_UPLOAD" +
-            "   AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.IN_PROGRESS)" +
-            " OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_UPLOAD" +
-            "   AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.VERIFICATION_PENDING)" +
-            " OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_VERIFICATION" +
-            "   AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.ACCEPTED)" +
-            " OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_VERIFICATION_FINAL" +
-            "   AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.IN_PROGRESS)" +
-            " OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_VERIFICATION_FINAL" +
-            "   AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.ACCEPTED)" +
-            " OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.CLIENT_EVALUATION" +
-            "   AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.IN_PROGRESS)" +
-            " OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.CLIENT_EVALUATION" +
-            "   AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.ACCEPTED)" +
-            " OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.PRESENCE_CHECK" +
-            "   AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.VERIFICATION_PENDING)"
+    @Query("""
+            SELECT id
+                FROM IdentityVerificationEntity id
+                WHERE ((id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_UPLOAD
+                    AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.IN_PROGRESS)
+                   OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_UPLOAD
+                    AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.VERIFICATION_PENDING)
+                   OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_VERIFICATION
+                    AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.ACCEPTED)
+                   OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_VERIFICATION_FINAL
+                    AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.IN_PROGRESS)
+                   OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.DOCUMENT_VERIFICATION_FINAL
+                    AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.ACCEPTED)
+                   OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.CLIENT_EVALUATION
+                    AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.IN_PROGRESS)
+                   OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.CLIENT_EVALUATION
+                    AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.ACCEPTED)
+                   OR (id.phase = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.PRESENCE_CHECK
+                    AND id.status = com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.VERIFICATION_PENDING))
+                   AND EXISTS (
+                       SELECT 1
+                       FROM id.documentVerifications documentVerification
+                       WHERE documentVerification.providerName = :documentVerificationProvider
+                   )
+            """
     )
-    Stream<IdentityVerificationEntity> streamAllIdentityVerificationsToChangeState();
-
+    Stream<IdentityVerificationEntity> streamAllIdentityVerificationsToChangeState(final String documentVerificationProvider);
 
     /**
      * Return identity verification IDs by the given process ID. Include only not yet finished entities.

--- a/enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java
+++ b/enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java
@@ -559,7 +559,7 @@ public class IdentityVerificationService {
      * @return identity verifications
      */
     public Stream<IdentityVerificationEntity> streamAllIdentityVerificationsToChangeState() {
-        return identityVerificationRepository.streamAllIdentityVerificationsToChangeState();
+        return identityVerificationRepository.streamAllIdentityVerificationsToChangeState(identityVerificationConfig.getDocumentVerificationProvider());
     }
 
     private void moveToDocumentUpload(final OwnerId ownerId, final IdentityVerificationEntity idVerification, final IdentityVerificationStatus status) {

--- a/enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/service/StateMachineService.java
+++ b/enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/service/StateMachineService.java
@@ -126,7 +126,7 @@ public class StateMachineService {
     @Transactional(readOnly = true)
     public void changeMachineStatesInBatch() {
         final AtomicInteger countFinished = new AtomicInteger(0);
-        try (Stream<IdentityVerificationEntity> stream = identityVerificationService.streamAllIdentityVerificationsToChangeState().parallel()) {
+        try (Stream<IdentityVerificationEntity> stream = identityVerificationService.streamAllIdentityVerificationsToChangeState()) {
               stream.filter(identityVerification -> identityVerification.getDocumentVerifications().stream()
                             .anyMatch(doc -> identityVerificationConfig.getDocumentVerificationProvider().equals(doc.getProviderName())))
                     .forEach(identityVerification -> {

--- a/enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/service/StateMachineService.java
+++ b/enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/service/StateMachineService.java
@@ -19,7 +19,6 @@ package com.wultra.app.onboardingserver.statemachine.service;
 import com.wultra.app.enrollmentserver.model.integration.OwnerId;
 import com.wultra.app.onboardingserver.common.database.entity.IdentityVerificationEntity;
 import com.wultra.app.onboardingserver.common.errorhandling.IdentityVerificationException;
-import com.wultra.app.onboardingserver.configuration.IdentityVerificationConfig;
 import com.wultra.app.onboardingserver.impl.service.IdentityVerificationService;
 import com.wultra.app.onboardingserver.statemachine.EnrollmentStateProvider;
 import com.wultra.app.onboardingserver.statemachine.consts.EventHeaderName;
@@ -68,8 +67,6 @@ public class StateMachineService {
     private final IdentityVerificationService identityVerificationService;
 
     private final TransactionTemplate transactionTemplate;
-
-    private final IdentityVerificationConfig identityVerificationConfig;
 
     @Transactional
     public StateMachine<OnboardingState, OnboardingEvent> processStateMachineEvent(OwnerId ownerId, String processId, OnboardingEvent event)
@@ -126,26 +123,24 @@ public class StateMachineService {
     @Transactional(readOnly = true)
     public void changeMachineStatesInBatch() {
         final AtomicInteger countFinished = new AtomicInteger(0);
-        try (Stream<IdentityVerificationEntity> stream = identityVerificationService.streamAllIdentityVerificationsToChangeState()) {
-              stream.filter(identityVerification -> identityVerification.getDocumentVerifications().stream()
-                            .anyMatch(doc -> identityVerificationConfig.getDocumentVerificationProvider().equals(doc.getProviderName())))
-                    .forEach(identityVerification -> {
-                        final String processId = identityVerification.getProcessId();
-                        final OwnerId ownerId = new OwnerId();
-                        ownerId.setActivationId(identityVerification.getActivationId());
-                        ownerId.setUserId(identityVerification.getUserId());
-                        logger.debug("Changing state of machine for process ID: {}", processId);
+        try (Stream<IdentityVerificationEntity> stream = identityVerificationService.streamAllIdentityVerificationsToChangeState().parallel()) {
+            stream.forEach(identityVerification -> {
+                final String processId = identityVerification.getProcessId();
+                final OwnerId ownerId = new OwnerId();
+                ownerId.setActivationId(identityVerification.getActivationId());
+                ownerId.setUserId(identityVerification.getUserId());
+                logger.debug("Changing state of machine for process ID: {}", processId);
 
-                        transactionTemplate.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
-                        transactionTemplate.executeWithoutResult(status -> {
-                            try {
-                                processStateMachineEvent(ownerId, processId, OnboardingEvent.EVENT_NEXT_STATE);
-                                countFinished.incrementAndGet();
-                            } catch (IdentityVerificationException e) {
-                                logger.warn("Unable to change state for process ID: {}", processId, e);
-                            }
-                        });
-                    });
+                transactionTemplate.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
+                transactionTemplate.executeWithoutResult(status -> {
+                    try {
+                        processStateMachineEvent(ownerId, processId, OnboardingEvent.EVENT_NEXT_STATE);
+                        countFinished.incrementAndGet();
+                    } catch (IdentityVerificationException e) {
+                        logger.warn("Unable to change state for process ID: {}", processId, e);
+                    }
+                });
+            });
         }
         if (countFinished.get() > 0) {
             logger.debug("Changed state of {} identity verifications", countFinished.get());

